src/common/models/data-cube/data-cube.ts

```diff
@@ -49,25 +52,21 @@ function formatTimeDiff(diff: number): string {
   return diff + ' days';
 }
 
-function checkUnique(dimensions: List<Dimension>, measures: List<Measure>, dataCubeName: string) {
-  var seenDimensions: Record<string, number> = {};
-  var seenMeasures: Record<string, number> = {};
+function checkUniquenessOfNamesAcrossDimensionsAndMeasures(dimensions: Dimensions, measures: Measures, dataCubeName: string) {
```

Comment:
Code style nitpick: consider spliting function declarations into multiple lines for better readablity.
ESLint provides linting for such rules. ;)